sdk/identity/azure-identity/inc/azure/identity/client_secret_credential.hpp

```diff
@@ -37,25 +62,21 @@ namespace Azure { namespace Identity {
      * @param tenantId Tenant ID.
      * @param clientId Client ID.
      * @param clientSecret Client Secret.
-     * @param authority Authentication authority URL to set. If omitted, initializes credential with
-     * default authority (Azure AD global authority - "https://login.microsoftonline.com/").
-     *
-     * @note Example of a \p authority string: "https://login.microsoftonline.us/". See national
-     * clouds' Azure AD authentication endpoints:
-     * https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud.
+     * @param options #TokenCredentialOptions.
      */
     explicit ClientSecretCredential(
         std::string tenantId,
         std::string clientId,
         std::string clientSecret,
-        std::string authority = g_aadGlobalAuthority)
+        TokenCredentialOptions options = TokenCredentialOptions())
```

Comment:
Per our guidelines this should be a ClientSecretCredentialOptions and the ClientSecretCredentialsOptions would derive from TokenCredentialOptions

Reply:
This is not how .NET has it, but I agree, ClientSecretCredentialOptions makes sense. I would not have TokenCredentialOptions for now then, because ClientSecretCredential is the only user so far. If needed, we can add TokenCredentialOptions later, and make ClientSecretCredentialOptions inherit it, without breaking change.

Reply:
> and make ClientSecretCredentialOptions inherit it, without breaking change.

I agree it won't be breaking. I wish we had a reference doc with some basic examples of when things are considered breaking or not, specifically in C++. @CaseyCarter - do you have a good reference?

Reply:
I think if the old code still compiles when you compile the new SDK lib, and your code, it is not considered breaking; not the ABI.